### PR TITLE
Add more comments to the Config structs

### PR DESCRIPTION
**Pull request by dgageot**
Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/schema/latest/config.go [hunk 1]
```diff
@@ -44,17 +54,44 @@ func (c *SkaffoldPipeline) GetVersion() string {
 
 // BuildConfig contains all the configuration for the build steps
 type BuildConfig struct {
+	// Artifacts lists the images you're going to be building
+	// you can include as many as you want here.
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: missing a comma or semicolon or new sentence here

File: pkg/skaffold/schema/latest/config.go [hunk 2]
```diff
@@ -77,28 +114,75 @@ type DateTimeTagger struct {
...
-	LocalBuild       *LocalBuild       `yaml:"local,omitempty" yamltags:"oneOf=build"`
+	// LocalBuild describes how to do a build on the local docker daemon
+	// and optionally push to a repository.
+	LocalBuild *LocalBuild `yaml:"local,omitempty" yamltags:"oneOf=build"`
+
+	// GoogleCloudBuild describes how to do a remote build on Google Cloud Build.
 	GoogleCloudBuild *GoogleCloudBuild `yaml:"googleCloudBuild,omitempty" yamltags:"oneOf=build"`
-	KanikoBuild      *KanikoBuild      `yaml:"kaniko,omitempty" yamltags:"oneOf=build"`
+
+	// KanikoBuild describes how to do a on-cluster build using
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: "an on-cluster..."

File: pkg/skaffold/schema/latest/config.go [hunk 3]
```diff
@@ -77,28 +114,75 @@ type DateTimeTagger struct {
...
+	// UseBuildkit use BuildKit to build Docker images.
+	UseBuildkit bool `yaml:"useBuildkit,omitempty"`
 }
 
-// GoogleCloudBuild contains the fields needed to do a remote build on
-// Google Cloud Build.
+// GoogleCloudBuild describes how to do a remote build on
+// [Google Cloud Build](https://cloud.google.com/cloud-build/docs/).
+// Docker and Jib artifacts can be built on Cloud Build. The `projectId` needs
+// to be provided and the currently logged user should be given permissions to trigger
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: logged in ?

File: pkg/skaffold/schema/latest/config.go [hunk 4]
```diff
@@ -118,23 +202,49 @@ type KanikoCache struct {
...
-	PullSecretName  string              `yaml:"pullSecretName,omitempty"`
-	Namespace       string              `yaml:"namespace,omitempty"`
-	Timeout         string              `yaml:"timeout,omitempty"`
-	Image           string              `yaml:"image,omitempty"`
-	DockerConfig    *DockerConfig       `yaml:"dockerConfig,omitempty"`
+	// BuildContext the Kaniko build context: `gcsBucket` or `localDir`.
+	// Defaults to `localDir`.
+	BuildContext *KanikoBuildContext `yaml:"buildContext,omitempty"`
+
+	// Cache
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: hmm

File: pkg/skaffold/schema/latest/config.go [hunk 5]
```diff
@@ -118,23 +202,49 @@ type KanikoCache struct {
...
+	Cache *KanikoCache `yaml:"cache,omitempty"`
+
+	// AdditionalFlags
+	AdditionalFlags []string `yaml:"flags,omitempty"`
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: hmm

File: pkg/skaffold/schema/latest/config.go [hunk 6]
```diff
@@ -178,30 +298,76 @@ type KubectlFlags struct {
...
+	Values map[string]string `yaml:"values,omitempty,omitempty"`
+
+	// Namespace the Kubernetes namespace.
+	Namespace string `yaml:"namespace,omitempty"`
+
+	// Version the version of the chart.
+	Version string `yaml:"version,omitempty"`
+
+	// SetValues a list of key-value pairs.
+	// If present, Skaffold will sent `--set` flag to Helm CLI and append all pairs after the flag.
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: will send

File: pkg/skaffold/schema/latest/config.go [hunk 7]
```diff
@@ -178,30 +298,76 @@ type KubectlFlags struct {
...
+
+	// Wait if `true`, Skaffold will send `--wait` flag to Helm CLI.
+	// Defaults to `false`.
+	Wait bool `yaml:"wait,omitempty"`
+
+	// RecreatePods if `true`, Skaffold will send `--recreate-pods` flag to Helm CLI.
+	// Defaults to `false`.
+	RecreatePods bool `yaml:"recreatePods,omitempty"`
+
+	// SkipBuildDependencies
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: hmm

File: pkg/skaffold/schema/latest/config.go [hunk 8]
```diff
@@ -234,59 +400,117 @@ type HelmConventionConfig struct {
...
+	// BuildTarget the `bazel build` target to run
+	// For example: "//:skaffold_example.tar"
+	BuildTarget string `yaml:"target,omitempty"`
+
+	// BuildArgs additional args to pass to `bazel build`.
+	// For example: ["arg1", "arg2"]
+	BuildArgs []string `yaml:"args,omitempty"`
 }
 
+// JibMavenArtifact builds containers using the Jib plugin for Maven.
```
Comment (jonjohnsonjr) [inline pkg/skaffold/schema/latest/config.go]: These are different style than BazelArtifact and DockerArtifact

**Comment by dgageot** [comment]
Thanks for the help @jonjohnsonjr!

**Pull request merged**